src/redux/books/booksSlice.js

```diff
@@ -1,41 +1,103 @@
+/* eslint-disable consistent-return */
+/* eslint-disable no-unused-vars */
 /* eslint-disable no-param-reassign */
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+import restructureApiResponse from '../../utils/restructureApiResponse';
+
+const baseAPI = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/ATwGr9HsrW4O0MQXnRyQ/books';
+
+export const getBooks = createAsyncThunk(
+  'books/getBooks',
+  // eslint-disable-next-line consistent-return
+  async (_, _thunkAPI) => {
+    try {
+      const res = await axios.get(baseAPI);
+      return res.data || [];
+    } catch (error) {
+      console.log(error);
+    }
+  },
+);
+
+export const addBook = createAsyncThunk(
+  'books/addBook',
+  async (newBook, thunkAPI) => {
+    try {
+      const res = await axios.post(baseAPI, newBook);
+      if (res.data === 'Created') {
+        return newBook;
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  },
+);
+
+export const removeBook = createAsyncThunk(
+  'books/removeBook',
+  async (id, thunkAPI) => {
+    try {
+      await axios.delete(`${baseAPI}/${id}`);
+      return id;
+    } catch (error) {
+      console.log(error);
```

Comment:
After using the `console.log` to debug and check errors in your code, kindly consider removing them before pushing to github. Do not commit console.log to your repo. It’s ugly, it kills performance and it can make confidential data be visible to anyone using the browser tools to look at your website. :wink: